whats-new-in-pgroll-0-10-0.mdx

```diff
@@ -0,0 +1,465 @@
+
+```json
+{
+  "create_table": {
+    "name": "employees",
+    "columns": [
+      {
+        "name": "email",
+        "varchar(254)"
+      }
+    ]
+  }
+},
+{
+  "create_constraint": {
+    "name": "my_unique_email",
+    "type": "unique",
+    "columns": ["email"]
+    "up": {
+      "email": "TODO"
+    }
+  }
+}
+
+```
+
+can be simplified to
+
+```json
+{
+  "create_table": {
+    "name": "employees",
+    "columns": [
+      {
+        "name": "email",
+        "varchar(254)"
+      }
+    ],
+    "constraints": [
+      {
+        "name": "my_unique_email",
+        "type": "unique",
+        "columns": ["email"]
+      }
+    ]
+  }
+}
+```
+
+Also, the same applies to some cases when the ORM does the backfilling for new columns. For example, when you add a new column with a default value, an ORM produces these SQL statements:
+
+```sql
+ALTER TABLE "adults" ADD COLUMN "age" smallint DEFAULT 18 NULL CHECK ("age" >= 18);
+ALTER TABLE "adults" ALTER COLUMN "level" DROP DEFAULT;
+
+UPDATE adults SET age = 18;
+
+ALTER TABLE adults ALTER COLUMN age SET NOT NULL;
+
+UPDATE "adults" SET "age" = 18 WHERE "age" IS NULL;
+SET CONSTRAINTS ALL IMMEDIATE;
+
+ALTER TABLE "adults" ALTER COLUMN "age" SET NOT NULL;
+```
+
+Resulting in the following pgroll migration using the `convert` subcommand:
+
+```json
+{
+  "add_column": {
+    "column": {
+      "check": {
+        "constraint": "age \u003e= 0",
+        "name": "age_check"
+      },
+      "default": "18",
+      "name": "age",
+      "nullable": true,
+      "type": "smallint"
+    },
+    "table": "adults",
+    "up": "TODO: Implement SQL data migration"
+  }
+},
+{
+  "alter_column": {
+    "column": "age",
+    "default": null,
+    "table": "adults",
+    "down": "TODO: Implement SQL data migration",
+    "up": "TODO: Implement SQL data migration"
+  }
+},
+{
+  "sql": {
+    "up": "update adults set age = 18"
+  }
+},
+{
+  "alter_column": {
+    "column": "age",
+    "nullable": false,
+    "table": "adults",
+    "up": "TODO: Implement SQL data migration"
+    "down": "TODO: Implement SQL data migration",
+  }
+},
+{
+  "sql": {
+    "up": "UPDATE \"adults\" SET \"age\" = 18 WHERE \"age\" IS NULL"
+  }
+},
+{
+  "sql": {
+    "up": "SET CONSTRAINTS ALL IMMEDIATE"
+  }
+},
+{
+  "alter_column": {
+    "column": "age",
+    "nullable": false,
+    "table": "adults",
+    "up": "TODO: Implement SQL data migration"
+    "down": "TODO: Implement SQL data migration",
+  }
+}
+```
+
+This can be written as a single pgroll migration:
+
+```json
+{
+  "add_column": {
+    "column": {
+      "check": {
+        "constraint": "age \u003e= 0",
```

Comment:
```suggestion
        "constraint": "age >= 0",
```

but again, shouldn't it be `18` rather than `0`?

Reply:
This was the output of pgroll, i just copy-pasted it. We can write it differently in the post imo.